packages/vuetify/src/composables/items.ts

```diff
@@ -0,0 +1,77 @@
+// Utilities
+import { computed } from 'vue'
+import { getObjectValueByPath, getPropertyFromItem, propsFactory } from '@/util'
+
+// Types
+import type { PropType } from 'vue'
+
+export interface InternalItem {
+  [key: string]: any
+  title: string
+  value: any
+  children?: InternalItem[]
+}
+
+export interface ItemProps {
+  items: (string | Partial<InternalItem>)[]
+  itemTitle: string
+  itemValue: any
+  itemChildren: string
+  itemProps: (item: any) => Partial<InternalItem>
+}
+
+// Composables
+export const makeItemsProps = propsFactory({
+  items: {
+    type: Array as PropType<ItemProps['items']>,
+    default: () => ([]),
+  },
+  itemTitle: {
+    type: String,
+    default: 'title',
+  },
+  itemValue: {
+    type: String,
+    default: 'value',
+  },
+  itemChildren: {
+    type: String,
+    default: 'children',
+  },
+  itemProps: {
+    type: Function as PropType<ItemProps['itemProps']>,
+    default: (item: any) => ({}),
+  },
+}, 'item')
+
+export function transformItem (props: ItemProps, item: any) {
+  const title = getPropertyFromItem(item, props.itemTitle, item)
+  const value = getPropertyFromItem(item, props.itemValue, title)
+  const children = getObjectValueByPath(item, props.itemChildren, [])
```

Comment:
Is there a case were you'd want `children: []` in the original item to be kept?

Reply:
Can we differentiate between **undefined** from an empty **array**?

Reply:
The current logic in treeview (and I think it's the same way in v2) is that no children property indicates leaf node. If the item has an empty children array, then it's still a branch node because it might contain lazy loaded items.

Reply:
What is the expected behavior if an item has **children** and that value is **undefined**? Can we use that in some way?